src/src/PluginStructs/P145_data_struct.cpp

```diff
@@ -0,0 +1,655 @@
+const struct P145_SENSORDEF sensorData[] PROGMEM =
+{
+  // User defined, open for experiments/own sensor definition
+  {     
+      "USER",       // Name
+      "unknown",    // gas
+      p145AlgNone,  // preferred/tuned algorithm
+      0.0,          // cleanRatio
+      0.0,          // PARA scaling factor value
+      0.0,          // PARB exponent value
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-135
+  {
+      "MQ-135",     // Name
+      "CO2",        // gas
+      p145AlgA,     // preferred/tuned algorithm
+      0.0,          // cleanRatio
+      116.6020682,  // PARA scaling factor value
+      2.769034857,  // PARB exponent value
+      0.00035,      // CORA
+      0.02718,      // CORB
+      1.39538,      // CORC
+      0.0018,       // CORD
+      -0.003333333, // CORE
+      -0.001923077, // CORF
+      1.130128205,  // CORG
+  },
+    // MQ-2
+  {
+      "MQ-2",       // Name
+      "H2",         // gas
+      p145AlgB,     // preferred/tuned algorithm
+      9.83,         // cleanRatio
+      987.99,       // PARA scaling factor
+      -2.162,       // PARB exponent value
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-3
+  {
+      "MQ-3",       // Name
+      "alcohol",    // gas
+      p145AlgB,     // preferred/tuned algorithm
+      60.0,         // cleanRatio
+      0.3934,       // PARA scaling factor
+      -1.504,       // PARB exponent value
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-4
+  {
+      "MQ-4",       // Name
+      "CH4",        // gas
+      p145AlgB,     // preferred/tuned algorithm
+      4.4,          // cleanRatio
+      1012.7,       // PARA scaling
+      -2.786,       // PARB exponent
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG
+  },
+  // MQ-7
+  {
+      "MQ-7",       // Name
+      "CO",         // gas
+      p145AlgB,     // preferred/tuned algorithm
+      27.5,         // cleanRatio
+      491204,       // PARA scaling
+      -5.826,       // PARB exponent
+      0.0,          // CORA
+      0.0,          // CORB
+      0.0,          // CORC
+      0.0,          // CORD
+      0.0,          // CORE
+      0.0,          // CORF
+      0.0,          // CORG   
+  }
+};
+const int nbrOfTypes = (int)(sizeof(sensorData) / sizeof(struct P145_SENSORDEF));
```

Comment:
Please mark this as a `constexpr`.
When marked as such, the compiler will compute it at compile time and simply use the result in the code instead of actually computing it at run-time.